### PR TITLE
WIP Mnt fixup after mds changes + API changes

### PR DESCRIPTION
**Pull request by tacaswell**
There are still some tests failing because I broke mds in ways that it's test suite does not catch, but dataportal does.

### CODE REVIEW

**Comment by tacaswell** [comment]
Also still need to re-work the dataportal internals to use the (still being finished) simplified mds api (`get_Y_given_X` for Y in documents and X in [uid, oid, runstart_uid, event_descritpor_uid])

**Comment by tacaswell** [comment]
Annoyingly, this PR is net-positive.....for now.

File: dataportal/broker/simple_broker.py [hunk 1]
```diff
@@ -308,14 +310,14 @@ def fill_event(event):
...
+            event.data[data_key] = fs.retrieve(value)
 
 
-class Header(Document):
+class Header(doc.Document):
     """A dictionary-like object summarizing metadata for a run."""
 
     @classmethod
-    def from_run_start(cls, run_start, verify_integrity=True):
+    def from_run_start(cls, run_start, verify_integrity=False):
```
Comment (ericdill) [inline dataportal/broker/simple_broker.py:317]: why did `verify_integrity` change from true to false?

File: dataportal/broker/simple_broker.py [hunk 2]
```diff
@@ -327,113 +329,131 @@ def from_run_start(cls, run_start, verify_integrity=True):
...
+                    mc.descriptors_by_start(run_start_uid)]
+
+    except mc.NoEventDescriptors:
+        ev_descs = []
+
+    header['descriptors'] = ev_descs
+    return doc.Document('header', header)
+
+
+def summerize_header(header):
```
Comment (ericdill) [inline dataportal/broker/simple_broker.py]: `summerize` -> `summarize`

Comment (ericdill) [inline dataportal/broker/simple_broker.py]: This function needs a docstring

File: dataportal/broker/simple_broker.py [hunk 3]
```diff
@@ -327,113 +329,131 @@ def from_run_start(cls, run_start, verify_integrity=True):
...
-            if k in special_keys:
-                continue
-            s.write("{0}: {1}".format(k, self[k]))
-
-        return s.readout()
+        return summerize_header(self)
+
+
+def make_header(run_start, allow_no_run_stop=False):
+    header = dict()
```
Comment (ericdill) [inline dataportal/broker/simple_broker.py]: Can you add a docstring to this function/

File: dataportal/broker/simple_broker.py [hunk 4]
```diff
@@ -327,113 +329,131 @@ def from_run_start(cls, run_start, verify_integrity=True):
...
+    # make sure that our run_start is really a document
+    # and get the uid
+    run_start_uid = mc.doc_or_uid_to_uid(run_start)
+    run_start = mc.run_start_given_uid(run_start_uid)
+    # fill in the run_start
+    header['run_start'] = run_start
+
+    # fields to copy out of run_start into top-level header
+    run_start_copy = {'start_time': 'time',
+                      'time': 'time',
```
Comment (ericdill) [inline dataportal/broker/simple_broker.py]: I dont understand why you are setting `run_start.time` to `header.time` and to `header.start_time`

File: dataportal/broker/simple_broker.py [hunk 5]
```diff
@@ -327,113 +329,131 @@ def from_run_start(cls, run_start, verify_integrity=True):
...
-            s.write("{0}: {1}".format(k, self[k]))
-
-        return s.readout()
+        return summerize_header(self)
+
+
+def make_header(run_start, allow_no_run_stop=False):
+    header = dict()
+    # make sure that our run_start is really a document
+    # and get the uid
```
Comment (ericdill) [inline dataportal/broker/simple_broker.py]: These two lines of comments don't really line up with the next two lines of code

**Comment by ericdill** [comment]
Maybe I missed this conversation, but when was it decided that **all** of the RunStart keys should not be copied into the header?

As of v0.1.0:

``` python
In [3]: print(hdr)

Header
======
beamline_config : None                                    
beamline_id     : csx                                     
group           :                                         
owner           :                                         
project         :                                         
run_start_uid   : 758bc8b2-b5f2-4cf5-a841-4490ccdd8c62    
sample          :
scan_id         : 5                                       
start_datetime  : 2015-09-09 16:46:18.113764              
start_time      : 1441831578.113764                       

  EventDescriptor
  ---------------
  +-----------------+--------+----------+-------+--------+
  | data keys       | dtype  | external | shape | source |
  +-----------------+--------+----------+-------+--------+
  | Tsam            | number |   None   |   []  | PV:pv3 |
  | linear_motor    | number |   None   |   []  | PV:pv1 |
  | scalar_detector | number |   None   |   []  | PV:pv2 |
  +-----------------+--------+----------+-------+--------+
  run_start       : 758bc8b2-b5f2-4cf5-a841-4490ccdd8c62    
  time            : 1441831578.1183035                      
  time_as_datetime: 2015-09-09 16:46:18.118303              
  uid             : 9133ff37-cd48-4091-b7d2-2d66477b5c5f    
```

As of commit 76e6ae6 in this PR

``` python
In [3]: hdr = db[-1]

In [4]: print(hdr)

header
======
exit_reason     : None                                    
exit_status     : None                                    
run_stop        : None                                    
sample          :
scan_id         : 5                                       
start_time      : 1441831578.113764                       
stop_time       : None                                    
time            : 1441831578.113764                       
uid             : 758bc8b2-b5f2-4cf5-a841-4490ccdd8c62    

  EventDescriptor
  ---------------
  +-----------------+--------+-------+--------+
  | data keys       | dtype  | shape | source |
  +-----------------+--------+-------+--------+
  | Tsam            | number |   []  | PV:pv3 |
  | linear_motor    | number |   []  | PV:pv1 |
  | scalar_detector | number |   []  | PV:pv2 |
  +-----------------+--------+-------+--------+
  run_start       : 758bc8b2-b5f2-4cf5-a841-4490ccdd8c62    
  time            : 1441831578.1183035                      
  uid             : 9133ff37-cd48-4091-b7d2-2d66477b5c5f    

  RunStart
  --------
  beamline_id     : csx                                     
  group           :                                         
  owner           :                                         
  project         :                                         
  sample          :
  scan_id         : 5                                       
  time            : 1441831578.113764                       
  uid             : 758bc8b2-b5f2-4cf5-a841-4490ccdd8c62    
```

File: dataportal/broker/simple_broker.py (hunk 1 above)
Comment (danielballan) [inline dataportal/broker/simple_broker.py:317]: I see that the function of `verify_integrity` has completely changed. The "not" of its value is passed to the `allow_no_run_stop` argument of `make_record`. Can we change `verify_integrity` to `allow_no_run_stop`, which is so much more explanatory? IMO it's an API break either way, given the complete change in meaning.

Obviously it makes sense that `allow_no_run_stop` should be True by default.

**Comment by danielballan** [comment]
@ericdill Yeah that doesn't look right. I take all of this as yet more indication that Header is not quite the right thing. I think we have overthought this.

**Comment by ericdill** [comment]
@tacaswell can you rebase this on master? Dan merged in some/many of these commits. It would be great to see what the diff is now

**Pull request closed without merging**